ai/src/main/java/org/triplea/ai/flowfield/FlowFieldAi.java

```diff
@@ -3,41 +3,174 @@
+                              frame.getMapPanel().clearTerritoryOverlay(territory);
+                            });
+                    frame.getMapPanel().repaint();
+                  }
+                }));
+    clearHeatmap.setSelected(true);
+    clearHeatmap.addItemListener(
+        e ->
+            actions.entrySet().stream()
+                .filter(entry -> entry.getKey() != e.getSource())
+                .forEach(entry -> entry.getValue().unselect()));
+    heatmapGroup.add(clearHeatmap);
+    heatmapMenu.add(clearHeatmap);
+
+    // add a button for each of the maps
+    diffusions.forEach(
+        (name, diffusion) -> {
+          final TerritoryDebugUiAction action =
+              new TerritoryDebugUiAction(frame, diffusion, getGameData().getMap());
+          final JRadioButtonMenuItem menuItem =
+              new JRadioButtonMenuItem(SwingAction.of(name, action));
+          actions.put(menuItem, action);
+          menuItem.addItemListener(
+              e ->
+                  actions.entrySet().stream()
+                      .filter(entry -> entry.getKey() != e.getSource())
+                      .forEach(entry -> entry.getValue().unselect()));
+          heatmapGroup.add(menuItem);
+          heatmapMenu.add(menuItem);
+        });
+
+    return List.of(
+        heatmapMenu,
+        new JMenuItem(SwingAction.of("Lanchester", new LanchesterDebugUiAction(frame, this))));
+  }
+
+  @Override
+  public void stopGame() {
+    super.stopGame();
+  }
+
   @Override
   protected void purchase(
       final boolean purchaseForBid,
       final int pusToSpend,
       final IPurchaseDelegate purchaseDelegate,
       final GameData data,
-      final GamePlayer player) {}
+      final GamePlayer player) {
+    if (isNewRound(getGameData().getCurrentRound())) {
+      calculateTurn();
+    }
+  }
+
+  private boolean isNewRound(final int round) {
+    if (round != this.round) {
+      this.round = round;
+      return true;
+    }
+    return false;
+  }
 
   @Override
   protected void tech(
-      final ITechDelegate techDelegate, final GameData data, final GamePlayer player) {}
+      final ITechDelegate techDelegate, final GameData data, final GamePlayer player) {
+    if (isNewRound(getGameData().getCurrentRound())) {
+      calculateTurn();
+    }
+  }
 
   @Override
   protected void move(
       final boolean nonCombat,
       final IMoveDelegate moveDel,
       final GameData data,
-      final GamePlayer player) {}
+      final GamePlayer player) {
+    if (isNewRound(getGameData().getCurrentRound())) {
+      calculateTurn();
+    }
+  }
 
   @Override
   protected void place(
       final boolean placeForBid,
       final IAbstractPlaceDelegate placeDelegate,
       final GameState data,
-      final GamePlayer player) {}
+      final GamePlayer player) {
+    if (isNewRound(getGameData().getCurrentRound())) {
+      calculateTurn();
+    }
+  }
+
+  private void calculateTurn() {}
```

Comment:
This is where diffusion maps will be calculated as well as the rest of the turn calculations.  The `@Override` methods from the ai player will just utilize the data that is generated in this method (and its helpers).

Reply:
Wouldn't be bad to either `throw new UnsupportedException("TODO")` or add a comment to this method body that the implementation will be landing. I'd recommend creating a Project so that you can have a project number that you can prefix such comments and then find them and clean them up.

Reply:
@DanVanAtta , I just tried to create a Project and it doesn't look like I have permissions to do that.  So could you create the project?

Reply:
https://github.com/triplea-game/triplea/projects/22

Let me know if you have any issues managing or adding items to the project